akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -200,6 +200,10 @@ object RejectionHandler {
         case MissingQueryParamRejection(paramName) ⇒
           complete((NotFound, "Request is missing required query parameter '" + paramName + '\''))
       }
+      .handle {
+        case InvalidRequiredValueForQueryParamRejection(paramName, requiredValue, actualValue) ⇒
+          complete((NotFound, s"Request is missing required value '$requiredValue' for query parameter '$paramName', actualValue '$actualValue'"))
```

Comment:
I think echo'ing a user-provided value in a response is considered a potential security risk (harmless on its own, but could be used in elaborate attach contexts), so perhaps best not to include the `actualValue`?

Reply:
actualValue is not included in error message anymore + I used new rejectRequestEntityAndComplete method to handle InvalidRequiredValueForQueryParamRejection